Panic when writing to GenericBuffer[any]

As the title states, when using a `GenericBuffer[any]` with a provided schema and attempting to write, the library panics with the following error:

```
# ./parquet_panic 
panic: reflect: call of reflect.Value.Field on interface Value

goroutine 1 [running]:
reflect.Value.Field({0x731d80?, 0xc0000172c0?, 0x10?}, 0x725620?)
	/usr/local/go/src/reflect/value.go:1266 +0xe5
github.com/segmentio/parquet-go.(*structField).Value(0x7?, {0x731d80?, 0xc0000172c0?, 0x1f?})
	/go/pkg/mod/github.com/segmentio/parquet-go@v0.0.0-20220914222423-67dbe8d21ca5/schema.go:439 +0x1d7
github.com/segmentio/parquet-go.deconstructFuncOfGroup.func1({0x0?, 0xc0000172c0?, 0x40ed3f?}, {0xf0?, 0x1d?, 0xc?}, {0x731d80?, 0xc0000172c0?, 0x782f80?})
	/go/pkg/mod/github.com/segmentio/parquet-go@v0.0.0-20220914222423-67dbe8d21ca5/row.go:486 +0x146
github.com/segmentio/parquet-go.(*Schema).Deconstruct(0xc00007e240, {0x0?, 0x0, 0x0}, {0x71bd00?, 0xc0000172c0?})
	/go/pkg/mod/github.com/segmentio/parquet-go@v0.0.0-20220914222423-67dbe8d21ca5/schema.go:232 +0x148
github.com/segmentio/parquet-go.(*GenericBuffer[...]).writeRows(0xc000018120, {0xc0000172c0, 0x1, 0x0})
	/go/pkg/mod/github.com/segmentio/parquet-go@v0.0.0-20220914222423-67dbe8d21ca5/buffer_go18.go:154 +0x1b2
github.com/segmentio/parquet-go.(*GenericBuffer[...]).Write(0xc0000c1f60?, {0xc0000172c0?, 0xc0000c1f70?, 0x408059?})
	/go/pkg/mod/github.com/segmentio/parquet-go@v0.0.0-20220914222423-67dbe8d21ca5/buffer_go18.go:125 +0x47
main.main()
	/parquet_panic/main.go:17 +0xf1
```

Minimal reproduction code:

```go
package main

import (
	parquet "github.com/segmentio/parquet-go"
)

type TestType struct {
	Key int
}

func main() {
	schema := parquet.SchemaOf(TestType{})
	buffer := parquet.NewGenericBuffer[any](schema)

	data := make([]any, 1)
	data[0] = TestType{Key: 0}
	_, _ = buffer.Write(data)
}
```

--- a/buffer_go18_test.go
+++ b/buffer_go18_test.go
@@ -186,3 +186,16 @@ func TestIssue343(t *testing.T) {
 	// must not panic
 	_ = parquet.NewGenericBuffer[any]()
 }
+
+func TestIssue346(t *testing.T) {
+	type TestType struct {
+		Key int
+	}
+
+	schema := parquet.SchemaOf(TestType{})
+	buffer := parquet.NewGenericBuffer[any](schema)
+
+	data := make([]any, 1)
+	data[0] = TestType{Key: 0}
+	_, _ = buffer.Write(data)
+}

--- a/schema.go
+++ b/schema.go
@@ -221,7 +221,7 @@ func (s *Schema) GoType() reflect.Type { return s.root.GoType() }
 // parquet schema.
 func (s *Schema) Deconstruct(row Row, value interface{}) Row {
 	v := reflect.ValueOf(value)
-	for v.Kind() == reflect.Ptr {
+	for v.Kind() == reflect.Ptr || v.Kind() == reflect.Interface {
 		if v.IsNil() {
 			v = reflect.Value{}
 			break
@@ -405,7 +405,7 @@ func (s *structNode) Fields() []Field {
 // reflect.Value if one of the fields was a nil pointer instead of panicking.
 func fieldByIndex(v reflect.Value, index []int) reflect.Value {
 	for _, i := range index {
-		if v = v.Field(i); v.Kind() == reflect.Ptr {
+		if v = v.Field(i); v.Kind() == reflect.Ptr || v.Kind() == reflect.Interface {
 			if v.IsNil() {
 				v = reflect.Value{}
 				break

